base/compiler/ssair/ir.jl

```diff
@@ -38,7 +38,7 @@ end
 
 block_for_inst(cfg::CFG, inst::Int) = block_for_inst(cfg.index, inst)
 
-function basic_blocks_starts(stmts::Vector{Any})
+@inline function basic_blocks_starts(stmts::Vector{Any})
```

Comment:
Out of curiosity, why is it a good idea to force inlining of such a large function?

Reply:
We can perform SROA optimization by replacing `BitSet` created by `basic_blocks_starts` with "scalar"-values (internal fields of `BitSet` in this case) if we inline `basic_blocks_starts` into `compute_basic_blocks`.

Ideally this inlining should be performed automatically with some advanced inlining cost model, but at this moment our inlining cost model only accounts for simple code complexity and doesn't reason about SROA-profitability at all. With EscapeAnalysis.jl merged this kind of optimization is now possible, but it still doesn't happen yet.